modules/fluid_properties/src/fluidproperties/LeadFluidProperties.C

```diff
@@ -250,21 +249,99 @@ LeadFluidProperties::T_from_v_e(Real v, Real e, Real & T, Real & dT_dv, Real & d
 }
 
 Real
-LeadFluidProperties::T_from_p_h(Real p, Real h) const
-{
-  auto lambda = [&](Real p, Real current_T, Real & new_h, Real & dh_dp, Real & dh_dT)
-  { h_from_p_T(p, current_T, new_h, dh_dp, dh_dT); };
-  Real T = FluidPropertiesUtils::NewtonSolve(
-               p, h, _T_initial_guess, _tolerance, lambda, name() + "::T_from_p_h")
-               .first;
-  // check for nans
-  if (std::isnan(T))
-    mooseError("Conversion from pressure (p = ",
-               p,
-               ") and enthalpy (h = ",
-               h,
-               ") to temperature failed to converge.");
-  return T;
+LeadFluidProperties::T_from_p_h(Real /*p*/, Real h) const
+{
+  // Obtained from sympy solving h(T) with T as a symbol
+  return -2067.9254113598 *
```

Comment:
in case you are wondering this beauty is very stable and accurate
![both](https://github.com/user-attachments/assets/de460b24-e748-47c3-b3cc-114cd5f70681)
![diff](https://github.com/user-attachments/assets/8c9a5f64-5f51-4ec7-b71f-69e5d2b340ec)